### PR TITLE
ENG-1274 Enhancements and UX Changes in Page Tree part of Page Designer

### PR DESCRIPTION
**Pull request by raxkaynan**

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **raxkaynan**

If they are not already assigned, you can assign the PR to them by writing `/assign @raxkaynan` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/ui/pages/config/ContentPagesContainer.js
Comment (gidesan) [inline src/ui/pages/config/ContentPagesContainer.js:86]: what `queryString` does when `value` is falsy?

File: src/ui/pages/config/ToolbarPageConfig.js
Comment (gidesan) [inline src/ui/pages/config/ToolbarPageConfig.js:19]: Please let's avoid adding `componentWillMount` since it's deprecated. Ensure your logic works with `componentDidMount `.

**Review by gidesan — CHANGES_REQUESTED** [review]
looks mostly good, only a few changes needed!

File: src/ui/pages/config/ToolbarPageConfig.js
Comment (raxkaynan) [reply src/ui/pages/config/ToolbarPageConfig.js:19]: Sorry I think you overlooked that it's `componentWillUnmount()`, not `componentWillMount()`.

File: src/ui/pages/config/ContentPagesContainer.js
Comment (raxkaynan) [reply src/ui/pages/config/ContentPagesContainer.js:86]: Yeah I'll probably put it in the `if` block as well to be clear.

File: src/ui/pages/config/ToolbarPageConfig.js
Comment (gidesan) [reply src/ui/pages/config/ToolbarPageConfig.js:19]: Correct, sorry!

**Review by gidesan — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/878#pullrequestreview-513356739" title="Approved">gidesan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [gidesan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
